### PR TITLE
Mkosi update helper

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

**Comment by DaanDeMeyer** [comment]
I would prefer to just rely on dependabot to do it for us tbh. No one is ever going to run this manually when there's a bot that will make a PR every once in a while to update it, so I think the end result will be that this won't really get used.

Even the distro script is something we don't run all that often

**Comment by keszybz** [comment]
So far we have 11 commits that bump the mkosi hash in since v256 and one more fixup commit for the case where it was bumped erroneously. Also 21 commits which update the distro hashes. Dependabot would require tags to be made, which seems unlikely. And the commit message produced by the scripts is vastly more useful than the dependabot generic dump. So I think both scripts will remain useful for the foreseeable future.

**Comment by DaanDeMeyer** [comment]
I don't mind merging this, but if dependabot makes a PR to update the hash, I'm just going to merge that instead of going through the effort of running this locally.

File: .github/workflows/mkosi.yml [hunk 1]
```diff
@@ -105,7 +105,7 @@ jobs:
 
     steps:
       - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332
-      - uses: systemd/mkosi@31b4e756c1484c302435653da5d3b9bdfae38518
+      - uses: systemd/mkosi@2c9954fa51a3a995bbdc02db6ef51f5bd27bc1ba
```
Comment (evverx) [inline .github/workflows/mkosi.yml:108]: I bumped mkosi along with systemd the other day and stuff like
```sh
‣ Setting BuildDirectory should be configured in [Build], not [Output].                                                                                       
‣ Setting CacheDirectory should be configured in [Build], not [Output].                                                                                       
‣ Setting Incremental should be configured in [Build], not [Host].                                                                                            
‣ Setting ToolsTreePackages should be configured in [Build], not [Host].                                                                                      
‣ Setting Include should be configured in [Include], not [Config].                                                                                            
‣ Setting Include should be configured in [Include], not [Config].                                                                                            
‣ Setting Include should be configured in [Include], not [Config].                                                                                            
‣ Setting Include should be configured in [Include], not [Config].              
```
started popping up. It's harmless but I'm curious how mkosi should be updated to be more or less sure that it works. I don't use it for anything other than the main branch of systemd so I kind of assumed that it should be fine to bump them together or should it bumped to commits from the mkosi job? It isn't hard of course but it's kind of annoying to keep track of those commits.

Comment (daandemeyer) [reply .github/workflows/mkosi.yml:108]: @evverx Bumping to the commit from the mkosi job would be safest. You could maybe write a little script to do that and contribute it to the tools/ folder. I do try to keep things mostly working though, as you said, the warnings are harmless.

Comment (evverx) [reply .github/workflows/mkosi.yml:108]: @DaanDeMeyer I'm not sure how to write a script that can be useful upstream. My scripts are weirdly specific and, for example, rely on the fact that mkosi is installed inside virtual environments. I think the only thing that can be useful in general would probably be a one-liner looking for the mkosi commit (and `fetch-mkosi.py` can already do that as far as I can see). I guess it should be possible to come up with a wrapper around mkosi that would fetch that commit and always use that instance of mkosi but I'm not going to use it anyway.

Maybe all those tips and tricks should just be documented somewhere in `docs/HACKING`. Currently it isn't obvious how to get it to work (I don't think `mkosi dependencies` are documented anywhere for example. It isn't even shown in `mkosi help`).

**Pull request merged**